sources/balanced_dollar.move

```diff
@@ -1,179 +1,202 @@
+        assert!(
+            method == CROSS_TRANSFER || method == CROSS_TRANSFER_REVERT, 
+            UnknownMessageType
+        );
 
-        assert!(method == CROSS_TRANSFER || method == CROSS_TRANSFER_REVERT, UnknownMessageType);
         if (method == CROSS_TRANSFER) {
-            assert!(from == vars.iconBnUSD, OnlyICONBnUSD);
-            // Messages.XCrossTransfer memory message = data.decodeCrossTransfer(); 
-            // (,string memory to) = message.to.parseNetworkAddress();
-            // _mint(to.parseAddress("Invalid account"), message.value);
+            assert!(from == network_address::from_string(config.icon_bnusd), OnlyICONBnUSD);
+            let message: XCrossTransfer = cross_transfer::decode(&msg);
+            let string_to = network_address::addr(&network_address::from_string(cross_transfer::to(&message)));
+            let to = address_from_hex_string(&string_to);
+            let amount: u64 = cross_transfer::value(&message);
+
+            coin::mint_and_transfer(get_treasury_cap_mut(carier),  amount, to, ctx)
         } else if (method == CROSS_TRANSFER_REVERT) {
-            assert!(from == vars.xCallNetworkAddress, OnlyCallService);
-            //Messages.XCrossTransferRevert memory message = data.decodeCrossTransferRevert();
-            //_mint(message.to, message.value);
-        } 
+            let message: XCrossTransferRevert = cross_transfer_revert::decode(&msg);
+            let to = cross_transfer_revert::to(&message);
+            let amount: u64 = cross_transfer_revert::value(&message);
+
+            coin::mint_and_transfer(get_treasury_cap_mut(carier),  amount, to, ctx)
+        };
+
+        xcall::execute_call_result(xcall,ticket,true,fee,ctx);
+    }
+
+    fun get_treasury_cap_mut<BALANCED_DOLLAR>(treasury_cap_carrier: &mut TreasuryCapCarrier<BALANCED_DOLLAR>): &mut TreasuryCap<BALANCED_DOLLAR> {
+        &mut treasury_cap_carrier.treasury_cap
     }
 
+    entry fun set_icon_bnusd(_: &AdminCap, config: &mut Config, icon_bnusd: String ){
+        config.icon_bnusd = icon_bnusd
+    }
+
+    entry fun set_xcall_network_address(_: &AdminCap, config: &mut Config, xcall_network_address: String ){
+        config.xcall_network_address = xcall_network_address
+    }
+
+    fun set_version(config: &mut Config, version: u64 ){
+        config.version = version
+    }
+
+    public fun get_version(config: &mut Config): u64{
+        config.version
+    }
+
+    entry fun migrate(_: &AdminCap, self: &mut Config) {
+        assert!(get_version(self) < CURRENT_VERSION, ENotUpgrade);
+        set_version(self, CURRENT_VERSION);
+    }
+    
 
     #[test_only]
-    public fun init_for_testing(ctx: &mut TxContext) {
-        init(BALANCED_DOLLAR {}, ctx);
+    public fun get_treasury_cap_for_testing<BALANCED_DOLLAR>(treasury_cap_carrier: &mut TreasuryCapCarrier<BALANCED_DOLLAR>): &mut TreasuryCap<BALANCED_DOLLAR> {
+        &mut treasury_cap_carrier.treasury_cap
     }
```

Comment:
we can rename this to network_id rather than network address. for SUI we can declare netId as constant "SUI" because we wont be deploying these in other chains.